sunpy/map/mapbase.py

```diff
@@ -730,16 +730,13 @@ def nickname(self):
     def nickname(self, n):
         self._nickname = n
 
-    @property
-    def date(self):
-        """
-        Image observation time.
+    def _get_date(self, key):
+        time = self.meta.get(key, None)
+        if time is None:
+            return
 
-        This is taken from the 'DATE-OBS' FITS keyword.
-        """
-        time = self.meta.get('date-obs', None)
         # Get the time scale
-        if time is not None and 'TAI' in time:
+        if 'TAI' in time:
```

Comment:
I think this bit of logic should remain solely in `.date`.

Reply:
Thinking about this, it should probably live in map sources? But given it's just raising a warning, I think worth keeping it for all the date fields?